src/utils/skylink.ts

```diff
@@ -0,0 +1,162 @@
+import base64 from "base64-js";
+import base32Encode from "base32-encode";
+import parse from "url-parse";
+import { CustomClientOptions } from "../client";
+import { trimForwardSlash, trimSuffix, trimUriPrefix } from "./string";
+
+/**
+ * Base custom options for methods hitting the API.
+ *
+ * @property [endpointPath] - The relative URL path of the portal endpoint to contact.
+ */
+export type BaseCustomOptions = CustomClientOptions & {
```

Comment:
The options stuff like `BaseCustomOptions` and `defaultOptions`. doesn't seem to fit in `skylink.ts`. Maybe `utils/options.ts`? `ParseSkylinkOptions` should stay here.